### PR TITLE
1.20.2 Port

### PR DESCRIPTION
**Pull request by ChrysanthCow**
Generally a port, but there are some changes that might be a little strange.

1. We can get rid of the Juuxel Vineflower plugin, as it is now built into FLoom.
2. Textured Widgets have been rewritten to not use mapped textures, hence why I separated the buttons. I'm unsure about the location of these textures, but they do have to be in the `gui/sprites` folder.
3. Displaying entities in GUIs with a method has also been rewritten, so I had to adjust a few values (mostly copied from the Horse like how it was previously). Quilt Mappings aren't finished for this method either, so you might notice some oddities with parameter names.
4. Updated `fabric` in dependencies to `fabric-api`.

### CODE REVIEW

File: src/main/java/dev/lambdaurora/lovely_snails/client/screen/SnailInventoryScreen.java [hunk 1]
```diff
@@ -111,7 +112,7 @@ public void closeScreen() {
 	/* Input */
 
 	@Override
-	public boolean mouseScrolled(double mouseX, double mouseY, double amount) {
+	public boolean mouseScrolled(double mouseX, double mouseY, double amount, double d) {
```
Comment (LambdAurora) [inline src/main/java/dev/lambdaurora/lovely_snails/client/screen/SnailInventoryScreen.java]:
```suggestion
	public boolean mouseScrolled(double mouseX, double mouseY, double scrollX, double scrollY) {
```

File: src/main/java/dev/lambdaurora/lovely_snails/client/screen/SnailInventoryScreen.java [hunk 2]
```diff
@@ -161,15 +162,18 @@ protected void drawBackground(GuiGraphics graphics, float delta, int mouseX, int
 		}
 
 		InventoryScreen.drawEntity(
-				graphics, x + 70, y + 60, 17,
-				(x + 51) - this.mouseX, (y + 75 - 50) - this.mouseY,
+				graphics, x + 40, y + 8,
+				x + 100,
+				y + 70, 17,
+				0.25F,
```
Comment (LambdAurora) [inline src/main/java/dev/lambdaurora/lovely_snails/client/screen/SnailInventoryScreen.java]:
```suggestion
				0.25f,
```

**Review by LambdAurora — COMMENTED** [review]
Please make sure the PNG files have been optimized out (with optipng/trimage) so when I merge I don't save in history the unoptimized versions of those files. (if not done already)

**Comment by ChrysanthCow** [comment]
> Please make sure the PNG files have been optimized out (with optipng/trimage) so when I merge I don't save in history the unoptimized versions of those files. (if not done already)

I haven't done this already, I'll do so in a bit.

Thanks for reminding me.

**Pull request merged**